src/app/Task.php

```diff
@@ -7,57 +7,63 @@
 use TaskForce\app\action\ActionDone;
 use TaskForce\app\action\ActionRefuse;
 use TaskForce\app\action\ActionRespond;
+use TaskForce\app\status\Status;
 use TaskForce\app\status\StatusCancel;
 use TaskForce\app\status\StatusDone;
 use TaskForce\app\status\StatusFailed;
 use TaskForce\app\status\StatusNew;
 use TaskForce\app\status\StatusProcess;
+use TaskForce\app\exception\BaseException;
 
 class Task
 {
-    private $errors = [];
-    private $userId = NULL;
-    private $employerId = NULL;
-    private $executorId = NULL;
-    private $currentStatus = NULL;
-
-    private $actionRespond = NULL;
-    private $actionRefuse = NULL;
-    private $actionDone = NULL;
-    private $actionCancel = NULL;
-
-    private $statusNew = NULL;
-    private $statusCancel = NULL;
-    private $statusProcess = NULL;
-    private $statusDone = NULL;
-    private $statusFailed = NULL;
-
-    public function __construct(int $userId, int $employerId, int $executorId) {
+    private array $errors = [];
+    private ?int $userId = NULL;
+    private ?int $employerId = NULL;
+    private ?int $executorId = NULL;
+    private ?Status $currentStatus = NULL;
+
+    private ?Action $actionRespond = NULL;
+    private ?Action $actionRefuse = NULL;
+    private ?Action $actionDone = NULL;
+    private ?Action $actionCancel = NULL;
+
+    private ?Status $statusNew = NULL;
+    private ?Status $statusCancel = NULL;
+    private ?Status $statusProcess = NULL;
+    private ?Status $statusDone = NULL;
+    private ?Status $statusFailed = NULL;
+
+    public function __construct(string $status, int $userId, int $employerId, ?int $executorId) {
         $this->userId = $userId;
         $this->employerId = $employerId;
         $this->executorId = $executorId;
 
         $this->initActions();
         $this->initStatuses();
-        $this->currentStatus = $this->statusNew;
+
+        if (!isset($this->getStatusesMap()[$status])) {
+            throw new BaseException("invalid status");
+        }
+        $this->currentStatus = $this->getStatusesMap()[$status];
     }
 
-    private function initActions() {
+    private function initActions():void {
```

Comment:
Если говорить про code style, то
```suggestion
    private function initActions(): void
     {
```